docs/modules/ROOT/pages/references/openstack/architecture.adoc

```diff
@@ -0,0 +1,145 @@
+:infra-type: OpenStack
+:infra-svg: ocp4-architecture-openstack.svg
+= APPUiO Managed OpenShift 4 on {infra-type}
+
+== Architecture overview
+
+include::partial$architecture/overview.adoc[]
+
+== {infra-type} requirements
+
+Red Hat OpenShift 4 imposes version requirements on the {infra-type} version.
+See the https://docs.redhat.com/en/documentation/openshift_container_platform/latest/html/installing_on_openstack/installing-openstack-installer-custom#prerequisites[upstream documentation] for the specific version requirements as well as further details on required {infra-type} requirements.
+For an overview of version compatibility, see the https://access.redhat.com/articles/4679401[OpenShift Container Platform on OpenStack Support Matrix].
+
+APPUiO Managed OpenShift 4 needs credentials to access the {infra-type} API for three main reasons:
+
+1. The OpenShift 4 installer needs access to {infra-type} to setup the OpenShift 4 cluster
+2. OpenShift 4 manages the VMs making up the cluster from within the cluster.
+3. The {infra-type} CSI driver manages additional block devices that can be used by applications
+
+
+
+== Networking
+
+=== Bastion host
+
+To deploy an APPUiO Managed OpenShift 4 cluster on {infra-type}, a bastion host inside the customer's premise is required.
+The bastion host:
+
+* must be accessible via SSH from a management system operated by VSHN
+* must have access to the {infra-type} API
+* must have unrestricted network access to the cluster's machine network
+* must run a recent Ubuntu version
+
+The bastion host is used to run the installer from, and for troubleshooting access to both the cluster and the vCenter.
+The bastion host must be provided by the {infra-type} infrastructure operator, but VSHN can handle management and maintenance.
+
+=== Machine network
+
+include::partial$architecture/networking-cluster.adoc[]
+
+=== Floating IPs
+
+To expose applications and the Kubernetes API outside the cluster, APPUiO Managed OpenShift 4 manages two floating IPs:
+
+1. The "API VIP" for the Kubernetes and OpenShift API.
+2. The "Ingress VIP" for the OpenShift Ingress Router.
+
+APPUiO Managed OpenShift 4 runs two `keepalived` instances to manage the API and ingress VIPs through VRRP.
+
+If applications should be exposed for non-HTTP(S) traffic (via `LoadBalancer` services), additional IPs in the machine network must be reserved to be used as VIPs.
+These additional VIPs will be managed by `keepalived` instances on the cluster.
```

Comment:
Not necessarily true for OpenStack. Afaik, if the OpenStack environment is provisioned with enough additional OpenStack floating IPs, LB services can be exposed on those via the OpenStack CCM.